ansible_testing/modules.py

```diff
@@ -181,7 +181,11 @@ def _just_docs(self):
             return False
 
     def _is_new_module(self):
-        return not module_loader.has_plugin(self.name)
+        if self.name.startswith("_"):
```

Comment:
Added in, for `ansible-modules-core` we now get

```
[DEPRECATION WARNING]: ec2_ami_search is kept for backwards compatibility but usage is discouraged. The module 
documentation details page may explain more about this rationale..
This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: docker is kept for backwards compatibility but usage is discouraged. The module documentation 
details page may explain more about this rationale..
This feature will be removed in a future release. Deprecation 
warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: nova_keypair is kept for backwards compatibility but usage is discouraged. The module 
documentation details page may explain more about this rationale..
This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: quantum_subnet is kept for backwards compatibility but usage is discouraged. The module 
documentation details page may explain more about this rationale..
This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: quantum_router_gateway is kept for backwards compatibility but usage is discouraged. The module 
documentation details page may explain more about this rationale..
This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: quantum_router_interface is kept for backwards compatibility but usage is discouraged. The module
 documentation details page may explain more about this rationale..
This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: quantum_floating_ip is kept for backwards compatibility but usage is discouraged. The module 
documentation details page may explain more about this rationale..
This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: keystone_user is kept for backwards compatibility but usage is discouraged. The module 
documentation details page may explain more about this rationale..
This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: glance_image is kept for backwards compatibility but usage is discouraged. The module 
documentation details page may explain more about this rationale..
This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: quantum_router is kept for backwards compatibility but usage is discouraged. The module 
documentation details page may explain more about this rationale..
This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: quantum_network is kept for backwards compatibility but usage is discouraged. The module 
documentation details page may explain more about this rationale..
This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: nova_compute is kept for backwards compatibility but usage is discouraged. The module 
documentation details page may explain more about this rationale..
This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: quantum_floating_ip_associate is kept for backwards compatibility but usage is discouraged. The 
module documentation details page may explain more about this rationale..
This feature will be removed in a future 
release. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
```
Still with a zero exit value. 

Which from looking at the code seems to be correct, i.e. we should be reporting deprecation notices.